tests/testsuite/patch.rs

```diff
@@ -3028,3 +3028,45 @@ foo v0.0.0 ([ROOT]/foo)
 
     assert_eq!(p.read_file("Cargo.lock"), p.read_file("Cargo.lock.orig"));
 }
+
+#[cargo_test]
+fn patch_with_base() {
+    let bar = project()
+        .at("bar")
+        .file("Cargo.toml", &basic_manifest("bar", "0.5.0"))
+        .file("src/lib.rs", "pub fn hello() {}")
+        .build();
+    Package::new("bar", "0.5.0").publish();
+
+    let p = project()
+        .file(
+            ".cargo/config.toml",
+            &format!(
+                "[path-bases]\ntest = '{}'",
+                bar.root().parent().unwrap().display()
+            ),
+        )
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.5.0"
+                authors = ["wycats@example.com"]
+                edition = "2018"
+
+                [dependencies]
+                bar = "0.5.0"
+
+                [patch.crates-io.bar]
+                path = 'bar'
+                base = 'test'
```

Comment:
nit: I would also find it easier to read for `base` to come before `path` as we are doing `base.join(path)`

Reply:
Switched.

Reply:
patch and the docs are still the other way.

Seeing the change, I'm mixed about what is "right" but being consistent would be good.

Reply:
ok